pkg/plugin/plugins.go

```diff
@@ -74,30 +74,26 @@ func (c *PluginsProcessor) Load() error {
 	if err != nil {
 		return fmt.Errorf("failed to load plugin dir: %v", err)
 	}
-	for _, f := range files {
-		// load shared object(.so) file
-		if filepath.Ext(f.Name()) == ".so" {
-			soFile := filepath.Join(common.DefaultTheClusterRootfsPluginDir(c.Cluster.Name), f.Name())
-			p, pt, err := c.loadOutOfTree(soFile)
-			if err != nil {
-				return err
-			}
-			Register(pt, p)
-		}
-		if yaml.Matcher(f.Name()) {
-			plugins, err := utils.DecodeCRDFromFile(filepath.Join(path, f.Name()), common.Plugin)
-			if err != nil {
-				return fmt.Errorf("failed to load plugin: %v", err)
+
+	if files != nil {
```

Comment:
if the files ==nil, 

the logic like `for _, f := range files` will not be executed actually

So no need to change that, just use `for _, f := range files` is better